server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/GlobalOrdinalsStringTermsAggregator.java

```diff
@@ -276,15 +227,14 @@ protected void doClose() {
                        SubAggCollectionMode collectionMode,
                        boolean showTermDocCountError,
                        Map<String, Object> metadata) throws IOException {
-            super(name, factories, valuesSource, order, format, bucketCountThresholds, null,
+            super(name, factories, a -> a.new StandardTermsResults(), valuesSource, order, format, bucketCountThresholds, null,
```

Comment:
Nit: maybe update the javadoc to say this can only be used for StandardTerms?

Reply:
:+1: